msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -166,21 +163,39 @@ public Integer getNumberOfDataPoints() {
 
   @Override
   public MsSpectrumType getSpectrumType() {
-    MsSpectrumType spectrumType = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mzValues,
-        intensityValues, getMzBinaryDataInfo().getArrayLength());
-    return spectrumType;
+    if (getCVValue(MzMLCV.cvCentroidSpectrum) != null)
+      return MsSpectrumType.CENTROIDED;
+
+    if (getCVValue(MzMLCV.cvProfileSpectrum) != null)
+      return MsSpectrumType.PROFILE;
+
+    return null;
   }
 
   @Override
   public Float getTIC() {
-    Float tic = MsSpectrumUtil.getTIC(intensityValues, getMzBinaryDataInfo().getArrayLength());
+    Float tic = null;
+    try {
+      tic = Float.valueOf(getCVValue(MzMLCV.cvTIC));
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    if (tic == null)
+      tic = MsSpectrumUtil.getTIC(getIntensityValues(), getMzBinaryDataInfo().getArrayLength());
```

Comment:
you have so many calls to getMzBinaryDataInfo().getArrayLength() throughout the code - how about saving the array length to a variable? It's not going to change once you read it.